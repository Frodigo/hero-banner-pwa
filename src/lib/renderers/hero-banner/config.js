import HeroBanner from './../../components/hero-banner'
import HeroBannerFigure from "../../components/hero-banner/hero-banner-figure";
import HeroBannerImage from "../../components/hero-banner/hero-banner-image";
import HeroBannerTape from "../../components/hero-banner/hero-banner-tape";
import HeroBannerHeading from "../../components/hero-banner/hero-banner-heading";
import HeroBannerAction from "../../components/hero-banner/hero-banner-button";

export const contentTypes = [
    {
        cssClass: 'hero-block',
        component: HeroBanner
    },
    {
        cssClass: 'hero-block__figure',
        component: HeroBannerFigure
    },
    {
        cssClass: 'hero-block__image',
        component: HeroBannerImage
    },
    {
        cssClass: 'hero-block__tape',
        component: HeroBannerTape
    },
    {
        cssClass: 'hero-block__heading',
        component: HeroBannerHeading
    },
    {
        cssClass: 'hero-block__action',
        component: HeroBannerAction
    },
];

/**
 * Get content type config by name.
 * @param contentType
 * @return {object}
 */
export function getContentType (contentType) {
    return contentTypes.find(item => item.cssClass === contentType);
}
