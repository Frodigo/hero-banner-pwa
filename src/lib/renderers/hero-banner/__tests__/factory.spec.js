import * as config from '../config';
import renderer from 'react-test-renderer';
import HeroBannerFactory from './../factory';
import React from 'react';

test('factory should render instance of content type', () => {
    const props = {
        data: {
            contentType: 'test',
            children: []
        }
    };
    const TestComponent = () => <div>Test Component</div>;
    config.getContentType = jest.fn().mockImplementation(contentType => {
        if (contentType === 'test') {
            return {
                component: TestComponent
            };
        }
    });
    const tree = renderer
        .create(<HeroBannerFactory {...props} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('factory should render all children content types', () => {
    const props = {
        data: {
            contentType: 'parent',
            children: [
                {
                    contentType: 'child',
                    children: []
                },
                {
                    contentType: 'child',
                    children: []
                }
            ]
        }
    };
    const ParentComponent = ({ children }) => (
        <div>
            Parent Component<div>{children}</div>
        </div>
    );
    const ChildComponent = () => <div>Child Component</div>;
    config.getContentType = jest.fn().mockImplementation(contentType => {
        if (contentType === 'parent') {
            return {
                component: ParentComponent
            };
        }
        if (contentType === 'child') {
            return {
                component: ChildComponent
            };
        }
    });

    const tree = renderer
        .create(<HeroBannerFactory {...props} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test('factory should not render content types that are not supported', () => {
    const props = {
        data: {
            contentType: 'broken',
            children: []
        }
    };
    const tree = renderer
        .create(<HeroBannerFactory {...props} />)
        .toJSON();

    expect(tree).toBeNull();
});
