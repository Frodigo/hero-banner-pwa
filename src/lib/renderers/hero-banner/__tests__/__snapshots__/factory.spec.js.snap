// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factory should render all children content types 1`] = `
<div>
  Parent Component
  <div>
    <div>
      Child Component
    </div>
    <div>
      Child Component
    </div>
  </div>
</div>
`;

exports[`factory should render instance of content type 1`] = `
<div>
  Test Component
</div>
`;
